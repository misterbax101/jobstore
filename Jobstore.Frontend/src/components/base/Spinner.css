.spinner {
    position: absolute;
    width: 100%;
    height: 100%;
    z-index: 1030;
    background: rgba(0,0,0,.5);
    opacity: .5;
}

.spinner .spinner-border{
    position: absolute;
    top:50%;
    left: 50%;
}