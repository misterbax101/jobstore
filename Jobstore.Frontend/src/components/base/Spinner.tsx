import React from 'react';
import { Row, Col, Spinner as BootsrapSpinner } from 'reactstrap';


const Spinner: React.FC = () =>
    (<Row className="text-center mt-5">
        <Col>
            <BootsrapSpinner></BootsrapSpinner>
        </Col>
    </Row>);

    export default Spinner;
