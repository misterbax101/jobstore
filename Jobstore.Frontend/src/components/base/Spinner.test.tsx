import React from 'react';
import { shallow } from 'enzyme';
import { Spinner as StrapSpinner  } from 'reactstrap';

import Spinner from './Spinner';

describe('<Spinner />', () => {

    it ('should render empty <Spinner />', () => {

         let wrapper =  shallow(<Spinner loading={false}/>)

         expect(wrapper.getElements()).toMatchSnapshot();
    });

    it ('should render <Spinner />', () => {

        let wrapper =  shallow(<Spinner loading={true}/>)

        expect(wrapper.getElements()).toMatchSnapshot();
        expect(wrapper.find(StrapSpinner).length).toBe(1);
   });
});
