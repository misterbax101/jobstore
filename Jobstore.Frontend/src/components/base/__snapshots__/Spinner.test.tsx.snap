// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Spinner /> should render <Spinner /> 1`] = `
Array [
  <div
    className="spinner"
  >
    <Spinner
      tag="div"
      type="border"
    >
      Loading...
    </Spinner>
  </div>,
]
`;

exports[`<Spinner /> should render empty <Spinner /> 1`] = `
Array [
  null,
]
`;
